From 3dacdad457b3d859993709f0181021ff0c2a865a Mon Sep 17 00:00:00 2001
From: IgnorantGuru <ignorantguru@gmx.com>
Date: Fri, 16 Oct 2015 09:20:53 -0600
Subject: [PATCH] fix misspelling priviledges #57

---
 po/udevil.pot | 6 +++---
 src/udevil.c  | 6 +++---
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/po/udevil.pot b/po/udevil.pot
index 43a2769..963c420 100644
--- a/po/udevil.pot
+++ b/po/udevil.pot
@@ -6,9 +6,9 @@
 #, fuzzy
 msgid ""
 msgstr ""
-"Project-Id-Version: udevil 0.4.3+\n"
+"Project-Id-Version: udevil 0.4.4+\n"
 "Report-Msgid-Bugs-To: \n"
-"POT-Creation-Date: 2015-04-28 18:39+0000\n"
+"POT-Creation-Date: 2015-10-16 15:19+0000\n"
 "PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
 "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
 "Language-Team: LANGUAGE <LL@li.org>\n"
@@ -20,7 +20,7 @@ msgstr ""
 #: ../src/udevil.c:616
 #, c-format
 msgid ""
-"udevil: error 1: unable to drop priviledges - please report this problem\n"
+"udevil: error 1: unable to drop privileges - please report this problem\n"
 msgstr ""
 
 #: ../src/udevil.c:810
diff --git a/src/udevil.c b/src/udevil.c
index bab80e9..ab4f5c1 100644
--- a/src/udevil.c
+++ b/src/udevil.c
@@ -39,7 +39,7 @@
 // environ
 #include <paths.h>
 
-// priviledges
+// privileges
 #include <sys/param.h>
 #include <sys/types.h>
 
@@ -547,7 +547,7 @@ void spc_sanitize_environment(int preservec, char **preservev) {
 }
 
 /* *************************************************************************
- * priviledges
+ * privileges
 ************************************************************************** */
 
 static int   orig_ngroups = -1;
@@ -613,7 +613,7 @@ void drop_privileges( int permanent )
 
     return;
 _drop_abort:
-    printf( _("udevil: error 1: unable to drop priviledges - please report this problem\n") );
+    printf( _("udevil: error 1: unable to drop privileges - please report this problem\n") );
     abort();
 }
 
